crates/sui-core/src/state_accumulator.rs

```diff
@@ -0,0 +1,588 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO(william) add metrics
+// TODO(william) make configurable
+
+use futures::stream::FuturesOrdered;
+
+use sui_types::base_types::ObjectDigest;
+use tracing::debug;
+use tracing::log::warn;
+
+use std::cmp::Ordering;
+use std::path::Path;
+use std::sync::Arc;
+
+use fastcrypto::hash::{Digest, MultisetHash};
+use mysten_metrics::spawn_monitored_task;
+use sui_types::accumulator::Accumulator;
+use sui_types::committee::EpochId;
+use sui_types::error::{SuiError, SuiResult};
+use sui_types::messages::TransactionEffects;
+use sui_types::messages_checkpoint::{
+    CheckpointSequenceNumber, CheckpointWatermark,
+};
+use tokio::sync::mpsc;
+use tokio_stream::StreamExt;
+use tokio::task::JoinHandle;
+use typed_store::rocks::{DBMap, TypedStoreError};
+use typed_store::traits::{TableSummary, TypedStoreDebug};
+use typed_store::Map;
+use typed_store_derive::DBMapUtils;
+
+use crate::authority::AuthorityState;
+use crate::checkpoints::CheckpointStore;
+
+const ROOT_STATE_HASH_KEY: u64 = 0;
+const MAX_CONCURRENCY: usize = 100;
+
+type AccumulatorTaskBuffer = FuturesOrdered<JoinHandle<SuiResult<(Accumulator, State)>>>;
+type EndOfEpochFlag = bool;
+
+#[derive(Debug, Clone)]
+pub struct State {
+    effects: Vec<TransactionEffects>,
+    end_of_epoch_flag: EndOfEpochFlag,
+    checkpoint_seq_num: CheckpointSequenceNumber,
+}
+
+#[derive(DBMapUtils)]
+pub struct StateAccumulatorStore {
+    // TODO: implement pruning policy as most of these tables can
+    // be cleaned up at end of epoch
```

Comment:
i think it may be better to move these tables to the epoch store - then you'll get cleanup for free.

Reply:
yeah was thinking about that as well